src/execution_plan/optimizations/utilize_indices.c

```diff
@@ -205,6 +205,13 @@ bool _applicableFilter(Index *idx, OpFilter *filter) {
 	const char **idx_fields = Index_GetFields(idx);
 	FT_FilterNode *filter_tree = filter->filterTree;
 
+	// Make sure the filter root is not a function.
```

Comment:
Move both `if(filter_tree->t == FT_N_EXP) {` and `if(FilterTree_containsOp(filter_tree, OP_NEQUAL)) {`
prior to `uint idx_fields_count = Index_FieldsCount(idx);` such that if the tree is not applicable we do not call `Index_FieldsCount` and `Index_GetFields`